src/readinglistbar.cpp

```diff
@@ -50,7 +50,17 @@ void ReadingListBar::setupList()
 void ReadingListBar::on_itemActivated(QListWidgetItem* item)
 {
     int index = ui->listWidget->row(item);
-    auto bookmark = KiwixApp::instance()->getLibrary()->getBookmarks().at(index);
+    auto library = KiwixApp::instance()->getLibrary();
+    auto bookmarks = library->getBookmarks();
+    QVector<kiwix::Bookmark> validBookmarks;
+    auto booksId = library->getBookIds();
+    for(auto& bookmark:bookmarks) {
+        if (booksId.contains(QString::fromStdString(bookmark.getBookId()))) {
+            validBookmarks.append(bookmark);
+        }
+    }
```

Comment:
I'm not sure we should have this in kiwix-desktop.
This is to the kiwix-lib to be sure that the content is coherent.
We could add a new argument to `kiwix::Library::getBookmarks` to only list valid bookmark. This way kiwix-desktop doesn't nothing special, when populating the list or reacting to a item activation.